app/javascript/graphs/segment_graphs.js

```diff
@@ -0,0 +1,109 @@
+import Highcharts from 'highcharts'
+import Exporting from 'highcharts/modules/exporting'
+Exporting(Highcharts)
+
+const buildSegmentGraphs = function(run) {
+  if (document.getElementById('segment-graphs') === null) {
+    return
+  }
+
+  const url = new URL(window.location.href)
+  const durationString = `${url.searchParams.get('timing') || 'real'}time_duration_ms`
+  const shortestString = `${url.searchParams.get('timing') || 'real'}time_shortest_duration_ms`
+
+  const pbData = []
+  const meanData = []
+  let resetCounter = run.attempts
+  const resetData = []
+
+  run.segments.forEach((segment, i) => {
+    let time = (segment[durationString] - segment[shortestString]) / 1000
+    if (time < 0) {
+      time = 0
+    }
+    pbData.push([segment.name, time])
+
+    time = 0
+    let counter = 0
+    if (segment.histories.length !== 0) {
+      segment.histories.forEach((attempt) => {
+        if (attempt[durationString] === 0) {
+          return
+        } else if (run.segments[i - 1] !== undefined) {
+          const prevSegAttempt = run.segments[i - 1].histories.find((prevAttempt) => {
+            return prevAttempt.attempt_number === attempt.attempt_number
+          })
+          if (prevSegAttempt !== undefined && prevSegAttempt[durationString] === 0) {
+            return
+          }
+        }
+
+        time += attempt[durationString]
+        counter++
+      })
+
+      time = ((time / counter) - segment[shortestString]) / 1000
+    }
+    meanData.push([segment.name, time])
+
+    resetData.push([segment.name, (1 - (segment.histories.length / resetCounter)) * 100])
+    resetCounter = segment.histories.length
```

Comment:
I see a couple of bugs on my runs with the reset bar.

On [my Sunshine Any%][1] splits, some bars are negative:
<img width="927" alt="screen shot 2018-12-29 at 12 19 58" src="https://user-images.githubusercontent.com/438911/50541900-73dcdc00-0b64-11e9-883d-b4102a0ce520.png">

On [my Portal Glitchless][2] splits, there's a 100% reset bar on my first segment, but I've reset 0 times in those splits (only 1 attempt and it was completed):

<img width="930" alt="screen shot 2018-12-29 at 12 19 49" src="https://user-images.githubusercontent.com/438911/50541913-d7ffa000-0b64-11e9-9d22-c1af93eb887d.png">

[1]: https://splits.io/1au
[2]: https://splits.io/3emj

Reply:
Just based on those screenshots, it might be good to have the grid lines very faintly resemble the color of the graph to which they refer, when applicable. E.g. the "75" grid line come from the Reset side would be slightly tinted green. Same with the labels. The more you can express without a legend, the easier graphs become to read I think.

Reply:
@glacials How old are those splits?  The bug comes from the fact that you have 1 attempt but no segment history for any of the segments, and if you make a brand new livesplit file it will have a history entry even if you only have the 1 attempt for the segment.

@jhobz That's a really good idea, thanks.  I was having a hard time trying to display all the legends while not just being a mess of information.

Reply:
Not sure when the Portal Glitchless splits were made, but I downloaded the blank splits file linked from Speedrun.com and ran my first and only attempt 12 days ago.

For the Sunshine splits, could it be that old versions of LiveSplit had absent segment histories for segments that weren't reached in that attempt? Not sure if I'm understanding that correctly though.

Reply:
For the sms one, your noki 4 split has 2 real history entries, while noki 5 has around 30 different ones.  So somehow it didn't track that you got to a previous split on the run but somehow still got past it.

For the portal one I'm really not sure what's up with that one then, but somehow it was fixed when I was fixing something else.